include/pybind11/embed.h

```diff
@@ -254,13 +254,15 @@ inline void finalize_interpreter() {
     if (builtins.contains(id) && isinstance<capsule>(builtins[id])) {
         internals_ptr_ptr = capsule(builtins[id]);
     }
+
+    Py_Finalize();
+
     // Local internals contains data managed by the current interpreter, so we must clear them to
     // avoid undefined behaviors when initializing another interpreter
+    // Must be cleared after Py_Finalize() so atexit and other hooks can use registered_types
```

Comment:
// Must be cleared only after Py_Finalize() so atexit and other hooks can still use use registered_types

?

Extra "only" and "still", to emphasize the importance.